brave/src/main/java/brave/internal/recorder/MutableSpan.java

```diff
@@ -117,6 +165,27 @@ public void tag(String key, String value) {
     pairs.add(value);
   }
 
+  public <T> void forEachTag(TagConsumer<T> tagConsumer, T target) {
+    for (int i = 0, length = pairs.size(); i < length; i += 2) {
+      Object first = pairs.get(i);
+      if (first instanceof Long) continue; // hit an annotation
+      tagConsumer.accept(target, first.toString(), pairs.get(i + 1).toString());
+    }
+  }
+
+  public <T> void forEachAnnotation(AnnotationConsumer<T> annotationConsumer, T target) {
+    for (int i = 0, length = pairs.size(); i < length; i += 2) {
+      Object first = pairs.get(i);
+      if (first instanceof String) continue; // hit a tag
```

Comment:
`instanceof` has some performance concern when you are using jre6(jre7 maybe)

Reply:
heh well it seems the reverse is true starting at java 8 (instanceof vs getClass ==)
https://stackoverflow.com/questions/103564/the-performance-impact-of-using-instanceof-in-java

Reply:
Ha, look like something is still right in the memory.